### PR TITLE
Tinkerer's Daemons that would be a waste of components cannot be made

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
tweak: Tinkerer's Daemons that would be useless cannot be made.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/clock_cult/clock_scripture.dm [hunk 1]
```diff
@@ -932,6 +932,29 @@ Judgement: 10 servants, 100 CV, and any existing AIs are converted or destroyed
...
+		return 0
+	return ..()
+
+/datum/clockwork_scripture/create_object/tinkerers_daemon/proc/check_waste()
+	var/servants = 0
+	for(var/mob/living/L in living_mob_list)
+		if(is_servant_of_ratvar(L))
+			servants++
+	if(!confirmed_waste && servants * 0.2 < clockwork_daemons+1)
+		var/confirm_wasteful = alert(user, "This daemon will prevent other daemons from working", "Do you want to waste components?", "Yes", "No")
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_scripture.dm]: please confirm you want to be an idiot asshole

File: code/game/gamemodes/clock_cult/clock_scripture.dm [hunk 2]
```diff
@@ -932,6 +932,29 @@ Judgement: 10 servants, 100 CV, and any existing AIs are converted or destroyed
...
+		return 0
+	return ..()
+
+/datum/clockwork_scripture/create_object/tinkerers_daemon/proc/check_waste()
+	var/servants = 0
+	for(var/mob/living/L in living_mob_list)
+		if(is_servant_of_ratvar(L))
+			servants++
+	if(!confirmed_waste && servants * 0.2 < clockwork_daemons+1)
+		var/confirm_wasteful = alert(user, "This daemon will prevent other daemons from working", "Do you want to waste components?", "No", "Yes")
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_scripture.dm]: please confirm you want to be an idiot asshole

File: code/game/gamemodes/clock_cult/clock_scripture.dm [hunk 3]
```diff
@@ -932,6 +932,32 @@ Judgement: 10 servants, 100 CV, and any existing AIs are converted or destroyed
...
+/datum/clockwork_scripture/create_object/tinkerers_daemon/proc/check_waste()
+	var/servants = 0
+	for(var/mob/living/L in living_mob_list)
+		if(is_servant_of_ratvar(L))
+			servants++
+	if(servants * 0.2 < clockwork_daemons)
+		invoker << "<span class='nezbere'>\"Daemons are already disabled, making [clockwork_daemons ? "more of them":"one"] would be a waste.\</span>"
+		return 0
+	if(!confirmed_waste && servants * 0.2 < clockwork_daemons+1)
+		var/confirm_wasteful = alert(invoker, "This daemon will prevent other daemons from working", "Do you want to waste components?", "No", "Yes")
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_scripture.dm]: please confirm you want to be an idiot asshole

**Comment by Supermichael777** [comment]
Just tell them they cant unless their is a non grief use

**Pull request merged**